### DISCUSSION TITLE
Desktop installation - exit service if app closed

### DISCUSSION THREAD

**Discussion by psflachglas** [rustdesk/rustdesk#7663] [General]
Hello

I have installed Rustdesk 1.2.3-2.
I have noticed that a Rustdesk service is now running in the background.
I do not want such a service to run in the background.
For me this is a security risk.

What is the reason that this service runs automatically after installation?
Unattended access? In my opinion this should not be activated automatically.
Is it not possible to start this service only as long as Rustdesk is running?

The following suggestion:
**Replace the stop button with a checkbox**

Settings -> General -> Service "Stop"

Enabled = Rustdesk service is always running
Disabled = Rustdesk service only runs when Rustdesk is open

Regards psflachglas

**Comment by rustdesk** [comment]
Not totally the same, but it can help you, https://twitter.com/rustdesk/status/1777703501671575941

**Reply by psflachglas** [reply]
What exactly happens with this option?
Could this be bypassed in case of a bug?

I have to assume that as long as Rustdesk is running, this provides an attack surface.
Therefore, I would strongly prefer not to have a service running in the background.
I should be able to assume that nobody can connect to my PC when the Rustdesk window is closed.
This should be the default behavior, in my opinion.
Don't get me wrong. I understand the need for a service.
It's great to have such an option.

Maybe you could implement something like my suggestion in the future.

Thank you very much.

**Reply by psflachglas** [reply]
Comment on nightly build "rustdesk-1.2.4-x86_64.exe":

**Note 1**
I have created a connection to the remote PC.
Then I closed the Rustdesk window on the remote PC.
An already connected connection is not interrupted when closing the window.

**Note 2**
The client can see from the green "LED" that the Rustdesk service is running. 
This already gives the attacker a valuable hint.

**Note 3**
The "Login Error" dialog is only displayed if the service is running on the remote PC.
If no service is running, "Connection is being established" is displayed on the client, followed by "Connection error"
Again this gives a valuable hint to an attacker

![2024-04-10 09_20_40-Dokument1  -  Kompatibilitätsmodus - Word](https://github.com/rustdesk/rustdesk/assets/94624822/c3773895-cdde-49ce-8c40-93e597f57c76)
